rundeckapp/grails-app/services/rundeck/services/ExecutionService.groovy

```diff
@@ -3697,15 +3697,19 @@ class ExecutionService implements ApplicationContextAware, StepExecutor, NodeSte
             )
 
             thread.start()
```

Comment:
if the exec.abortby is set already, why start the thread at all?

Reply:
I had problem not starting the thread (the steps stays forever in Waiting status) but now I tested again and it looks like those problems were caused by Intellij using debug mode and some stuck threads, now it works ok. I'm going to push the fix without starting it.